src/map/achievement.c

```diff
@@ -1862,9 +1868,11 @@ static void do_init_achievement(bool minimal)
 
 	VECTOR_INIT(achievement->rank_exp);
 
-	/* Read LibConfig Files */
-	achievement->readdb();
-	achievement->readdb_ranks();
+	if (battle_config.feature_enable_achievement == 1) {
```

Comment:
I think it's safer to always load these, even if the system is disabled (since the setting can be toggled at runtime with a `@reloadbattleconf`), similarly to what we do with the attendance DB

Reply:
removed this part and rebased